src/idl_gen_julia.cpp

```diff
@@ -0,0 +1,820 @@
+        case BASE_TYPE_STRING:
+          GenStringField(struct_def, field, code_ptr, has_defaults);
+          break;
+        case BASE_TYPE_STRUCT:
+        case BASE_TYPE_VECTOR:
+        case BASE_TYPE_UNION:
+          GenDependentField(struct_def, field, code_ptr, has_defaults, imports_ptr);
+          break;
+        default: FLATBUFFERS_ASSERT(0);
+      }
+    }
+  }
+
+  static std::string GenImports(const std::set<std::string> &imports) {
+    std::string impstr = "";
+    for (auto it = imports.begin(); it != imports.end(); ++it)
+      impstr += "import " + *it + "\n";
+    return impstr;
+  }
+
+  // Generate structs/tables
+  void GenObject(const StructDef &struct_def, std::string *code_ptr) {
+    if (struct_def.generated) return;
+
+    // always need FlatBuffers package for structs
+    std::set<std::string> imports;
+    imports.insert(JuliaPackageName);
+
+    bool has_defaults = false;
+
+    // generate all the fields
+    std::vector<std::string> offsets;
+
+    GenComment(struct_def.doc_comment, code_ptr, &JuliaCommentConfig);
+    for (auto it = struct_def.fields.vec.begin();
+         it != struct_def.fields.vec.end(); ++it) {
+      auto &field = **it;
+      if (field.deprecated) continue;
+      GenField(struct_def, field, code_ptr, &imports, &has_defaults);
+      offsets.push_back("0x" + IntToStringHex(field.value.offset, 8));
+    }
+    EndObject(struct_def, offsets, code_ptr);
+
+    // need to call BeginObject after EndObject because we don't know
+    // the defaults until we've looked at all the fields.
+    std::string struct_begin;
+    BeginObject(struct_def, &struct_begin, has_defaults);
+
+    *code_ptr = GenImports(imports) + "\n" + struct_begin + *code_ptr;
+
+    // Generate a functions for constructing the object from a buffer
+    NewObjectFromBuffer(struct_def, code_ptr);
+  }
+
+  void GenUnion(const EnumDef &enum_def, std::string *code_ptr) {
+    if (enum_def.generated) return;
+
+    // always need FlatBuffers package for unions
+    std::set<std::string> imports;
+    imports.insert(JuliaPackageName);
+    std::string union_name = NormalizedName(enum_def);
+    BeginUnion(union_name, code_ptr);
+    for (auto it = enum_def.vals.vec.begin(); it != enum_def.vals.vec.end();
+         ++it) {
+      auto &ev = **it;
+      std::string type_name = GenTypeGet(ev.union_type, &enum_def);
+      GenComment(ev.doc_comment, code_ptr, &JuliaCommentConfig);
+      UnionMember(type_name, code_ptr);
+      // special case, instead make every Union a union with Nothing
+      if (ev.name == "NONE") continue;
+      AddDependency(enum_def, ev.union_type, &imports);
+    }
+    EndUnion(code_ptr);
+
+    *code_ptr = GenImports(imports) + "\n" + *code_ptr;
+  }
+
+  void GenEnum(const EnumDef &enum_def, std::string *code_ptr) {
+    if (enum_def.generated) return;
+    GenComment(enum_def.doc_comment, code_ptr, &JuliaCommentConfig);
+    std::string enum_name = NormalizedName(enum_def);
```

Comment:
use `auto` where possible, here and elsewhere.

Reply:
Done!